snuba/datasets/errors_replacer.py

```diff
@@ -537,38 +535,106 @@ def process_merge(
     )
 
 
+def _convert_hash_list(
+    hashes: Sequence[str], state_name: ReplacerState, convert_types: bool = False
+) -> str:
+    if state_name == ReplacerState.ERRORS:
+        if convert_types:
+            return ", ".join(
+                "toUUID('%s')" % str(uuid.UUID(_hashify(h))) for h in hashes
+            )
+        else:
+            return ", ".join("'%s'" % str(uuid.UUID(_hashify(h))) for h in hashes)
+    else:
+        if convert_types:
+            return ", ".join("toFixedString('%s', 32)" % _hashify(h) for h in hashes)
+        else:
+            return ", ".join("'%s'" % _hashify(h) for h in hashes)
+
+
+def _build_grouping_hashes_filter(
+    message: Mapping[str, Any], state_name: ReplacerState
+) -> Optional[Tuple[List[str], List[str], MutableMapping[str, str]]]:
+    hashes = message.get("hashes") or []
+    hierarchical_hashes = message.get("hierarchical_hashes") or {}
+    if not hashes and not hierarchical_hashes:
+        return None
+
+    assert all(isinstance(h, str) for h in hashes)
+    assert all(isinstance(h, str) for h in hierarchical_hashes.values())
+
+    prewhere = []
+    where = []
+    query_args = {}
+
+    if hashes:
```

Comment:
Is it possible to have both `hashes` and `hierarchical_hashes`? If so, do we need both filters?

Reply:
we will not use both filters within the same message.